returns/instances.py

```diff
@@ -17,3 +24,28 @@ def is_io(container: BaseContainer) -> bool:
 
     """
     return isinstance(container, (IO, IOResult, RequiresContextIOResult))
+
+
+def is_result(container: BaseContainer) -> bool:
+    """
+    Verifies if a container is `` Result`` type.
+
+    .. code:: python
+
+      >>> from returns.maybe import Maybe
+      >>> from returns.result import Success
+
+      >>> assert is_result(Success(1.0)) is True
+      >>> assert is_result(Maybe.from_value(1.0)) is False
+
+    """
+    result_containers = (
+        Result,
+        FutureResult,
+        ContextResult,
+        ContextIOResult,
+        IOResult,
+        RequiresContextResult,
+        RequiresContextIOResult,
+    )
+    return isinstance(container, result_containers)
```

Comment:
I have some nit.
I think a better name for the variable is `result_container_type` than `result_containers` and in this case maybe the best not to use a temporary variable for one call and just use:
```
  return isinstance(container, (
    Result,
    FutureResult,
    ContextResult,
    ContextIOResult,
    IOResult,
    RequiresContextResult,
    RequiresContextIOResult,
  ))
```
Sorry, if this remark is meaningless.

Reply:
I agree with the variable's name, but not with removing the variable!
Using the variable it's more readable than passing all possible types to `isinstance` function.